client.go

```diff
@@ -89,13 +117,27 @@ func (client *Client) Handler(h http.Handler) http.Handler {
 		h = http.DefaultServeMux
 	}
 	fn := func(w http.ResponseWriter, r *http.Request) {
+		ctx := r.Context()
+		ctx = client.SetContext(ctx, map[string]interface{}{
+			"path":           r.URL.String(),
```

Comment:
Adding this data is great, but we'll want to send it as `cgi_data` (with special keys) instead of context--that's what we use to display request data in error reports. For an example, search for "cgi_data" on this page: https://docs.honeybadger.io/api/exceptions.html#sample-payload

Here's how we construct it in Elixir, for instance: https://github.com/honeybadger-io/honeybadger-elixir/blob/94c6c1f29cef70ae754bf12aa99779849ddd0b70/lib/honeybadger/plug_data.ex#L49

I'd probably hold off on including this in the current PR (since it's kind of a separate feature) and then submit a subsequent PR after we get the new `ctx` API worked out.